src/models.py

```diff
@@ -1,6 +1,28 @@
+import re
 from enum import Enum
 
+from web3 import Web3
+
 
 class Network(Enum):
     MAINNET = 'mainnet'
     GCHAIN = 'gchain'
+
+
+class Account:
+
+    def __init__(self, address: str):
+        if Account._is_valid(address):
+            self.address = Web3.toChecksumAddress(address)
+        else:
+            raise ValueError(f"Invalid Ethereum Address {address}")
+
+    def __str__(self):
+        return self.address
+
+    @staticmethod
+    def _is_valid(address: str) -> bool:
+        if re.match(r'^(0x)?[0-9a-f]{40}$', address, flags=re.IGNORECASE):
```

Comment:
Why not simply return the value of `re.match()`?

Reply:
I wasn't sure if the result of this is Boolean or not. I think I could do re.match is not None if that looks cleaner.

Reply:
`is not None` would seem cleaner to me. Except this would cause the line to so long that the linter starts complaining.

Reply:
Fixed!